dm/pkg/binlog/event/util.go

```diff
@@ -408,6 +408,27 @@ func GetCharsetCodecByStatusVars(statusVars []byte) (encoding.Encoding, error) {
 	}
 }
 
+// GetTimezoneByStatusVars returns the timezone of upstream for "datetime" type alter table commands
+// For "ALTER TABLE ADD x datetime", not for "ALTER TABLE ADD x timestamp".
+func GetTimezoneByStatusVars(statusVars []byte, upstreamTZStr string) (string, error) {
+	vars, err := statusVarsToKV(statusVars)
+	b, ok := vars[QTimeZoneCode]
+
+	if !ok {
+		if err == nil {
```

Comment:
this part of code is quite anti-pattern, seems imitate logic in getSQLMode(GetServerCollationByStatusVars too). it's introduced in https://github.com/pingcap/dm/pull/1176/files, to fix some undocumented var which may causes syncer stop. 

we should leave a comment to explain why we didn't check error first. or better to refactor this part of code, we don't care about undocumented vars anyway, it should not return error.

Reply:
fired an issue https://github.com/pingcap/tiflow/issues/6121

Reply:
I have modified this part of the code. Please take a look at whether it corresponds to what you meant.